### PR TITLE
Add documentation of QualifiedDo to the user's guide

### PR DESCRIPTION
**Pull request by Tritlo**

### CODE REVIEW

File: docs/users_guide/exts/qualified_do.rst [hunk 1]
```diff
@@ -14,9 +14,11 @@ Qualified do-notation
 
     Allow the use of qualified ``do`` notation.
 
+``QualifiedDo`` enables qualifying a ``do`` block with a module name, to control which operations to use for
+the monadic combinators that the ``do`` notation desugars to.
 
-When ``-XQualifiedDo`` is activated, it allows you to *qualify* do notation
-with a module name, as demonstrated below: ::
+With ``-XQualifiedDo`` is enabled, you can *qualify* the ``do`` notation by writing ``modid.do``, where
```
Comment (facundominguez) [inline docs/users_guide/exts/qualified_do.rst]:
```suggestion
When ``-XQualifiedDo`` is enabled, you can *qualify* the ``do`` notation by writing ``modid.do``, where
```

**Review by facundominguez — APPROVED** [review]
Looks good!

In addition, I'd like to have some paragraph near the beginning, motivating why someone might want to use -XQualifiedDo.

File: docs/users_guide/exts/qualified_do.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+     M.do { rec { x1 <- u1; … ; xn <- un }; stmts }  =
+       M.do
+       { (x1, …, xn) <- M.mfix (\~(x1, …, xn) -> M.do { x1 <- u1; …; xn <- un; M.return (x1, …, xn)})
+       ; stmts
+       }
+
+If a name ``M.op`` is required by the desugaring process (and only if it's required!) but the name is
+not in scope, an error message is produced:
+
+  “Desugaring statement ``<stmt>`` requires ``<M.op>`` which is not in scope”
```
Comment (facundominguez) [inline docs/users_guide/exts/qualified_do.rst]:
```suggestion
not in scope, it is reported as an error.
```

The error text does not match closely the implementation, and it is prone to get outdated if we try to copy it.

File: docs/users_guide/exts/qualified_do.rst [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+The qualified operations are subject to the same type restrictions as their counterparts with ``-XRebindableSyntax``.
```
Comment (facundominguez) [inline docs/users_guide/exts/qualified_do.rst]: The documentation of RebindableSyntax does not highlight what these restrictions are. I thought I had removed this statement from the proposal. I would omit it here as well.

Comment (Tritlo) [reply docs/users_guide/exts/qualified_do.rst]: I've removed it!

**Comment by Tritlo** [comment]
I moved (and rewrote) a motivating example to the top, before all the syntax stuff. You think it is enough?

**Comment by facundominguez** [comment]
Thanks @Tritlo. The example at the beginning is useful. I contributed a couple of paragraphs as well.

**Pull request merged**